web/experiments/feature_view.py

```diff
@@ -0,0 +1,283 @@
+
+    return features_f1_scored, features_params
+
+
+def _generate_figure(fs_id, fig_title, norms_path='./CSLB',
+                     fig_path='./cslb_feature_view_{:%d-%m-%Y_%H:%M}.png'.format(datetime.datetime.now()),
+                     show_visual=False):
+    """
+    This function generate picture for feature categories according to CLSB classification
+
+    Parameters
+    ----------
+    fs_id: dict
+        Keys are features and F1-score are values for logreg trained for it
+
+    norms_path: string
+        Path to cslb dat files
+
+    fig_path: string
+    Path where figure is saved. If is None figure is not stored
+
+    fig_title: string
+        added to title after CSLB
+
+    show_visual: bool
+        show swarmplot using matplotlib
+    """
+
+    # for feature fit
+    categ = pd.read_csv(norms_path, sep='\t')
+    categ = categ[['feature type', 'concept', 'feature']]
+
+    def get_feature_category(word_str):
+        """Get all feature category"""
+        word_str = word_str.replace('_', ' ')
+        return categ.loc[categ['feature'] == word_str]['feature type'].unique().astype(str)
+
+    def binsort_features(feature_f1):
+        """Partition by feature type eg. 'encycopedic'"""
+        cat_val = defaultdict(list)
+        for f in feature_f1.keys():
+            for c in get_feature_category(f):
+                cat_val[c].append(feature_f1[f])
+
+        return cat_val
+
+    bf = binsort_features(fs_id)
+
+    df = pd.DataFrame.from_dict(bf, orient='index')
+
+    logging.info('Current matplotlib backend is {}'.format(plt.get_backend()))
+
+    # workaround for running matplotlib image generation without Xserver
+    if not show_visual:
+        logging.info('Switching matplotlib backend to Agg')
+        plt.switch_backend('agg')
+
+    sns.set_style("whitegrid")
+
+    tdf = df.transpose()
+    tdf.sort_index(axis=1, inplace=True)
+
+    sns.boxplot(data=tdf, boxprops=dict(alpha=0.2))
+    ax = sns.swarmplot(data=tdf)
+    plt.title('CSLB {}'.format(fig_title))
+
+    ax.set(xlabel='Feature category', ylabel='Feature fit score')
+
+    if fig_path is not None:
+        ax.get_figure().savefig(fig_path)
+        logging.info('Figure saved: {}'.format(fig_path))
+
+    if show_visual:
+        plt.show()
+
+    plt.close()
+
+
+def _store_data(fs_id, fp_id, file_path):
+    score_df = pd.DataFrame(list(fs_id.items()), columns=['Feature', 'F1score'])
+
+    params_data = {'Feature': fp_id[:, 0], 'Alpha': fp_id[:, 1], 'EndIter': fp_id[:, 2], 'Intercept': fp_id[:, 3],
+                   'Coefs': list(fp_id[:, 4:])}
+
+    params_df = pd.DataFrame(params_data, columns=['Feature', 'Alpha', 'EndIter', 'Intercept', 'Coefs'])
+
+    df = score_df.merge(right=params_df, on='Feature')
+
+    df = df.set_index('Feature')
+    df.sort_index(inplace=True)
+    logging.info('Saving: {}'.format(file_path))
+    df.to_csv(file_path)
+
+
+def figure_from_csv(path, fig_title):
```

Comment:
A bit picky, but please change name to generate_figure (and u can call internally to "_generate_figure")